Add back openssl 1.1.1

Shelnutt2 opened the pull request:
There are some downstream packages i.e TileDB-VCF, which depends on htslib, that still need openssl 1.1.1. htslib hasn't had a rebuild on bioconda channel in a few month and is still tied to 1.1.1.

<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

xylar commented:
@Shelnutt2 are there downstream packages on conda-forge still using python 3.7?  I thought support had been dropped.  Or do you mean on other channels?

Shelnutt2 commented:
> @Shelnutt2 are there downstream packages on conda-forge still using python 3.7? I thought support had been dropped. Or do you mean on other channels?

I typed up the wrong message from my debugging. Its not actually python 3.7 but its packages from bioconda, such as htslib (which is used by TileDB-VCF) that haven't been rebuilt in months and still are linked against openssl 1.1.1. I've adjusted the description and updated the commit message.

Shelnutt2 commented:
Here is a basic example in a fresh env showing htslib pulls in openssl 1.1.1.

```
conda install -c conda-forge -c bioconda htslib
WARNING: The conda.compat module is deprecated and will be removed in a future release.
Collecting package metadata: done
Solving environment: - 
The environment is inconsistent, please check the package plan carefully
The following packages are causing the inconsistency:

  - defaults/linux-64::asn1crypto==0.24.0=py37_0
  - defaults/linux-64::certifi==2019.3.9=py37_0
  - defaults/linux-64::cffi==1.11.5=py37he75722e_1
  - defaults/linux-64::chardet==3.0.4=py37_1
  - defaults/linux-64::conda==4.6.11=py37_0
  - defaults/linux-64::cryptography==2.4.2=py37h1ba5d50_0
  - defaults/linux-64::idna==2.8=py37_0
  - defaults/linux-64::libedit==3.1.20181209=hc058e9b_0
  - defaults/linux-64::libffi==3.2.1=hd88cf55_4
  - defaults/linux-64::libgcc-ng==8.2.0=hdf63c60_1
  - defaults/linux-64::ncurses==6.1=he6710b0_1
  - defaults/linux-64::openssl==1.1.1b=h7b6447c_1
  - defaults/linux-64::pip==18.1=py37_0
  - defaults/linux-64::pycosat==0.6.3=py37h14c3975_0
  - defaults/linux-64::pycparser==2.19=py37_0
  - defaults/linux-64::pyopenssl==18.0.0=py37_0
  - defaults/linux-64::pysocks==1.6.8=py37_0
  - defaults/linux-64::python==3.7.3=h0371630_0
  - defaults/linux-64::readline==7.0=h7b6447c_5
  - defaults/linux-64::requests==2.21.0=py37_0
  - defaults/linux-64::ruamel_yaml==0.15.46=py37h14c3975_0
  - defaults/linux-64::setuptools==40.6.3=py37_0
  - defaults/linux-64::six==1.12.0=py37_0
  - defaults/linux-64::sqlite==3.27.2=h7b6447c_0
  - defaults/linux-64::tk==8.6.8=hbc83047_0
  - defaults/linux-64::urllib3==1.24.1=py37_0
  - defaults/linux-64::wheel==0.32.3=py37_0
  - defaults/linux-64::xz==5.2.4=h14c3975_4
  - defaults/linux-64::yaml==0.1.7=had09818_2
  - defaults/linux-64::zlib==1.2.11=h7b6447c_3
done


==> WARNING: A newer version of conda exists. <==
  current version: 4.6.11
  latest version: 23.1.0

Please update conda by running

    $ conda update -n base -c defaults conda



## Package Plan ##

  environment location: /usr/local

  added / updated specs:
    - htslib


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    _libgcc_mutex-0.1          |      conda_forge           3 KB  conda-forge
    _openmp_mutex-4.5          |            2_gnu          23 KB  conda-forge
    bzip2-1.0.8                |       h7f98852_4         484 KB  conda-forge
    c-ares-1.18.1              |       h7f98852_0         113 KB  conda-forge
    ca-certificates-2022.9.24  |       ha878542_0         150 KB  conda-forge
    htslib-1.16                |       h6bc39ce_0         2.3 MB  bioconda
    krb5-1.19.4                |       h568e23c_0         1.4 MB
    ld_impl_linux-64-2.39      |       hc81fddc_0         759 KB  conda-forge
    libcurl-7.87.0             |       h91b91d3_0         354 KB
    libdeflate-1.13            |       h166bdaf_0          79 KB  conda-forge
    libedit-3.1.20221030       |       h5eee18b_0         191 KB
    libev-4.33                 |       h516909a_1         104 KB  conda-forge
    libffi-3.4.2               |       h7f98852_5          57 KB  conda-forge
    libgcc-ng-12.2.0           |      h65d4601_19         931 KB  conda-forge
    libgomp-12.2.0             |      h65d4601_19         455 KB  conda-forge
    libnghttp2-1.46.0          |       hce63b2e_0         795 KB
    libnsl-2.0.0               |       h7f98852_0          31 KB  conda-forge
    libsqlite-3.40.0           |       h753d276_0         791 KB  conda-forge
    libssh2-1.10.0             |       haa6b8db_3         234 KB  conda-forge
    libuuid-2.32.1             |    h7f98852_1000          28 KB  conda-forge
    libzlib-1.2.13             |       h166bdaf_4          64 KB  conda-forge
    ncurses-6.3                |       h27087fc_1        1002 KB  conda-forge
    openssl-1.1.1s             |       h166bdaf_0         2.1 MB  conda-forge
    pip-22.3.1                 |     pyhd8ed1ab_0         1.5 MB  conda-forge
    python-3.11.0              |h582c2e5_0_cpython        35.9 MB  conda-forge
    readline-8.1.2             |       h0f457ee_0         291 KB  conda-forge
    setuptools-65.5.1          |     pyhd8ed1ab_0         731 KB  conda-forge
    tk-8.6.12                  |       h27826a3_0         3.3 MB  conda-forge
    tzdata-2022f               |       h191b570_0         118 KB  conda-forge
    wheel-0.38.4               |     pyhd8ed1ab_0          32 KB  conda-forge
    xz-5.2.6                   |       h166bdaf_0         409 KB  conda-forge
    zlib-1.2.13                |       h166bdaf_4          92 KB  conda-forge
    ------------------------------------------------------------
                                           Total:        54.6 MB

The following NEW packages will be INSTALLED:

  _libgcc_mutex      conda-forge/linux-64::_libgcc_mutex-0.1-conda_forge
  _openmp_mutex      conda-forge/linux-64::_openmp_mutex-4.5-2_gnu
  bzip2              conda-forge/linux-64::bzip2-1.0.8-h7f98852_4
  c-ares             conda-forge/linux-64::c-ares-1.18.1-h7f98852_0
  htslib             bioconda/linux-64::htslib-1.16-h6bc39ce_0
  krb5               pkgs/main/linux-64::krb5-1.19.4-h568e23c_0
  ld_impl_linux-64   conda-forge/linux-64::ld_impl_linux-64-2.39-hc81fddc_0
  libcurl            pkgs/main/linux-64::libcurl-7.87.0-h91b91d3_0
  libdeflate         conda-forge/linux-64::libdeflate-1.13-h166bdaf_0
  libev              conda-forge/linux-64::libev-4.33-h516909a_1
  libgomp            conda-forge/linux-64::libgomp-12.2.0-h65d4601_19
  libnghttp2         pkgs/main/linux-64::libnghttp2-1.46.0-hce63b2e_0
  libnsl             conda-forge/linux-64::libnsl-2.0.0-h7f98852_0
  libsqlite          conda-forge/linux-64::libsqlite-3.40.0-h753d276_0
  libssh2            conda-forge/linux-64::libssh2-1.10.0-haa6b8db_3
  libuuid            conda-forge/linux-64::libuuid-2.32.1-h7f98852_1000
  libzlib            conda-forge/linux-64::libzlib-1.2.13-h166bdaf_4
  tzdata             conda-forge/noarch::tzdata-2022f-h191b570_0

The following packages will be UPDATED:

  ca-certificates    pkgs/main::ca-certificates-2019.1.23-0 --> conda-forge::ca-certificates-2022.9.24-ha878542_0
  libedit                           3.1.20181209-hc058e9b_0 --> 3.1.20221030-h5eee18b_0
  libffi                 pkgs/main::libffi-3.2.1-hd88cf55_4 --> conda-forge::libffi-3.4.2-h7f98852_5
  libgcc-ng           pkgs/main::libgcc-ng-8.2.0-hdf63c60_1 --> conda-forge::libgcc-ng-12.2.0-h65d4601_19
  ncurses                 pkgs/main::ncurses-6.1-he6710b0_1 --> conda-forge::ncurses-6.3-h27087fc_1
  openssl              pkgs/main::openssl-1.1.1b-h7b6447c_1 --> conda-forge::openssl-1.1.1s-h166bdaf_0
  pip                   pkgs/main/linux-64::pip-18.1-py37_0 --> conda-forge/noarch::pip-22.3.1-pyhd8ed1ab_0
  python                 pkgs/main::python-3.7.3-h0371630_0 --> conda-forge::python-3.11.0-h582c2e5_0_cpython
  readline               pkgs/main::readline-7.0-h7b6447c_5 --> conda-forge::readline-8.1.2-h0f457ee_0
  setuptools         pkgs/main/linux-64::setuptools-40.6.3~ --> conda-forge/noarch::setuptools-65.5.1-pyhd8ed1ab_0
  tk                         pkgs/main::tk-8.6.8-hbc83047_0 --> conda-forge::tk-8.6.12-h27826a3_0
  wheel              pkgs/main/linux-64::wheel-0.32.3-py37~ --> conda-forge/noarch::wheel-0.38.4-pyhd8ed1ab_0
  xz                         pkgs/main::xz-5.2.4-h14c3975_4 --> conda-forge::xz-5.2.6-h166bdaf_0
  zlib                    pkgs/main::zlib-1.2.11-h7b6447c_3 --> conda-forge::zlib-1.2.13-h166bdaf_4
```

xylar commented:
Okay, I see.  Does bioconda have something like conda-forge's migrators to help automatically rebuild these packages so we don't need to support too many variants here?

Shelnutt2 commented:
> Okay, I see. Does bioconda have something like conda-forge's migrators to help automatically rebuild these packages so we don't need to support too many variants here?

It does not seem so. Recently we've also had problems with libdeflate, so the issues are growing. @jdblischak or @ihnorton do you happen to know if bioconda has any plans in the works to support migrations and auto-rebuilding.

jdblischak commented:
> do you happen to know if bioconda has any plans in the works to support migrations and auto-rebuilding.

I can't speak for their future plans, but I am familiar with the current procedure of how they update pinnings and migrate recipes. It's a much more manual, centralized process compared to the conda-forge migrations.

* First the pinning is updated in [`bioconda_utils-conda_build_config.yaml`](https://github.com/bioconda/bioconda-utils/blob/master/bioconda_utils/bioconda_utils-conda_build_config.yaml)
* Then a bioconda core maintainer migrates/rebuilds all the downstream recipes that depend on the pinned software in the [bulk branch](https://bioconda.github.io/developer/bulk.html)
* Once all the packages have been rebuilt and uploaded to the bioconda channel, the bulk branch is merged into master
* Here's a recent example: https://github.com/bioconda/bioconda-utils/pull/797#issuecomment-1227451182

jdblischak commented:
Quick update: the openssl migration issue has started to be discussed in bioconda https://github.com/bioconda/bioconda-recipes/issues/39356. The solution is known: they need to update the version of conda-forge-pinning they build their recipes against (currently [`2022.08.25.15.20.42`](https://github.com/bioconda/bioconda-utils/blob/25c91e552b5c3e583fa04203c8589372ec2a574c/bioconda_utils/bioconda_utils-requirements.txt#L26)). However, it's not clear to me what the timeline is for this potential migration
